packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -127,19 +128,23 @@ export class PendingStateManager implements IDisposable {
 			this.initialMessages.isEmpty(),
 			0x2e9 /* "Must call getLocalState() after applying initial states" */,
 		);
-		const newSavedOps = [...this.savedOps].filter(
-			(message) => message.referenceSequenceNumber >= (snapshotSequenceNumber ?? 0),
-		);
+		const newSavedOps = [...this.savedOps].filter((message) => {
+			assert(
+				message.sequenceNumber !== undefined,
+				"saved op should already have a sequence number",
+			);
+			return message.sequenceNumber >= (snapshotSequenceNumber ?? 0);
```

Comment:
This is probably minor, but what is the semantics of snapshotSequenceNumber === undefined?
Is it just staging of the changes (and it will become non-optional soon), or it's actually a state we support?
It's not very clear what are the contract around it, i.e. is it guaranteed that snapshot sequence number will not change through rehydration process if it was undefined on save?

Reply:
It's a state we support. If snapshotSequenceNumber is undefined, means snapshot refresh is disabled or we didn't refresh before capturing state again, so no filtering should be applied.

Reply:
But should we always pass sequence number of a snapshot?

Reply:
Basically, comparison with zero is wrong (or rather - not very strong). I'd rather see here a sequence number of a snapshot we will use - all the ops have to be higher than that sequence number, no matter if we use refresh or not